ports/esp32/machine_i2c.c

```diff
@@ -32,14 +32,28 @@
 
 #include "driver/i2c.h"
 
-#define I2C_0_DEFAULT_SCL (GPIO_NUM_18)
-#define I2C_0_DEFAULT_SDA (GPIO_NUM_19)
+#ifndef MICROPY_HW_I2C0_SCL
+#define MICROPY_HW_I2C0_SCL (GPIO_NUM_18)
+#endif
+
+#ifndef MICROPY_HW_I2C0_SDA
```

Comment:
Similar to how it's done in `machine_hw_spi.c` (and for stm32 periphs) it's simpler to just require "all or nothing" config: if a board defines `MICROPY_HW_I2C0_SCL` then it must also define `SDA`.  So this #if is not needed.

Reply:
@dpgeorge updated to depend on each `CLK` variable like with SPI. 9 lines lighter ⭐